### DISCUSSION TITLE
exit code at the right is gone

### DISCUSSION THREAD

**Discussion by Hr46ph** [romkatv/powerlevel10k#2770] [General]
I think a recent update borked this up, or perhaps something got spooked on  my end.

I ran the reconfigure a few times and I think I need to pick full frame with 2 lines, otherwise the right part of the prompt simply won't display.

When I manually edit .p10k.zsh I can get it back without full frame if I pick something that is always there, like time. If I don't, the right part completely disappears and won't show execution time or exit codes.

Is this a known issue, any ideas?

**Comment by romkatv** [comment]
Please attach your `.p10k.zsh` as a file (you'll need to rename it to `p10k.txt` or similar, or github won't let you attach it), describe what you do, what you expect, and what actually happens.

**Comment by Hr46ph** [comment]
[p10k.txt](https://github.com/user-attachments/files/17268571/p10k.txt)
All the hashed out lines at `POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS` are made during troubleshooting the disappearing of the right part of the prompt. Initially, that was left default.

I run `p10k configure` and pick the following options:
y, y, n, y, 1, y, 2, 1, 3, n, 3, 3, 5, 2, 1, 1, 2, 2, 2, n, 1, y

After reloading, only the left part of the prompt is visible, with the `>` on the second line.
<img width="1146" alt="image" src="https://github.com/user-attachments/assets/12327b80-0a37-45eb-8542-90b38ae187b4">

<img width="1146" alt="image" src="https://github.com/user-attachments/assets/28adbdbc-c9bc-4b45-a341-03a652f0594f">

Just noticed that when `vim` exists, it shows `took 12s`. Hadn't noticed that before. Adding `load` and source it, it starts showing the right part permanently.
<img width="1146" alt="image" src="https://github.com/user-attachments/assets/60a21f9a-73e3-467d-a3ba-45cfde0f9ec6">

What I expected, is that the right part of the prompt permanently shows. It used to with full frame and showed exit code every prompt.

**Reply by romkatv** [reply]
Everything on your screenshot looks as expected: right prompt is shown when there is something to show. For example, "took 12s".

If you want "boring" exit codes to be displayed, open `~/.p10k.zsh`, search for `POWERLEVEL9K_STATUS_ERROR` and set its value to `true`. As the comment above that parameter says, the default is to not show boring exit codes when you have `prompt_char` in prompt.

**Reply by Hr46ph** [reply]
Thanks for the quick responses, appreciate it!

So the only difference is that I previously used full frame which always shows the right hand part, it always shows it even when there is nothing to display.
<img width="782" alt="image" src="https://github.com/user-attachments/assets/7afdd25a-d8b0-4eb7-ac1c-97d46b40063a">

I wanted to move away from the full frame since copy/pasting commands with output brings along a line of spaces with the unicode frame. I didn't expect to loose the right hand part on the first line of the prompt as well (when there was nothing valueable to show).

I'll see if I can muck around with the settings to have it always show, as with the full frame.

**Reply by romkatv** [reply]
When you choose prompt with left frame, there is no `prompt_char` in it by default. Without `prompt_char` it is necessary to display `status` at all times, for without it you won't know whether the previous command has succeeded or not. `prompt_char` is very useful, while frame is functionally useless.
